nemo_aligner/algorithms/ppo.py

```diff
@@ -287,27 +393,41 @@ def compute_global_rollout_metrics(self, rollout_batches):
     def run_validation(self):
         self.model.prepare_for_inference()
 
-        num_val_micro_batches = compute_num_rollout_microbatches(self.val_dataloader)
-        val_dataloader = iter(self.val_dataloader)
+        _, rollout_metrics, _ = self._run_inference(
+            self.val_dataloader_builder, consumed_samples=0, is_validation=True
+        )
 
-        _, rollout_metrics = self._run_inference(val_dataloader, num_val_micro_batches, is_validation=True)
         self.model.finish_inference()
         return rollout_metrics
 
     @torch.no_grad()
-    def generate_rollouts(self, dataloader_iter, num_microbatches):
+    def generate_rollouts(self):
+        timing_metrics = {}
+
+        self.timer.start("prepare_for_inference")
         self.model.prepare_for_inference()
+        timing_metrics["prepare_for_inference"] = self.timer.stop_and_get_time("prepare_for_inference")
 
-        rollout_batches, rollout_metrics = self._run_inference(dataloader_iter, num_microbatches, is_validation=False)
+        rollout_batch, rollout_metrics, timer_metrics = self._run_inference(
+            self.train_dataloader_builder, consumed_samples=self.consumed_samples, is_validation=False
+        )
+
+        self.consumed_samples += rollout_metrics["consumed_samples"]
 
-        ppo_rollout_data, ppo_rollout_metrics = map(cpu_dict, self.generate_ppo_data(rollout_batches))
+        ppo_rollout_data, ppo_rollout_metrics = self.generate_ppo_data(rollout_batch)
+        ppo_rollout_metrics = cpu_dict(ppo_rollout_metrics)
```

Comment:
Seems useless since `generate_ppo_data()` already returns `cpu_dict(ppo_rollout_metrics)`

Reply:
ah good point, fixed in 8e2282d1daaf8898e101395dd7e99439a8e32e44